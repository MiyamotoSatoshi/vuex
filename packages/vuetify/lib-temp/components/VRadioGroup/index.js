import VRadioGroup from './VRadioGroup';
import VRadio from './VRadio';
export { VRadioGroup, VRadio };
export default {
    $_vuetify_subcomponents: {
        VRadioGroup,
        VRadio
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WUmFkaW9Hcm91cC9pbmRleC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLFdBQVcsTUFBTSxlQUFlLENBQUE7QUFDdkMsT0FBTyxNQUFNLE1BQU0sVUFBVSxDQUFBO0FBRTdCLE9BQU8sRUFBRSxXQUFXLEVBQUUsTUFBTSxFQUFFLENBQUE7QUFFOUIsZUFBZTtJQUNiLHVCQUF1QixFQUFFO1FBQ3ZCLFdBQVc7UUFDWCxNQUFNO0tBQ1A7Q0FDRixDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFZSYWRpb0dyb3VwIGZyb20gJy4vVlJhZGlvR3JvdXAnXG5pbXBvcnQgVlJhZGlvIGZyb20gJy4vVlJhZGlvJ1xuXG5leHBvcnQgeyBWUmFkaW9Hcm91cCwgVlJhZGlvIH1cblxuZXhwb3J0IGRlZmF1bHQge1xuICAkX3Z1ZXRpZnlfc3ViY29tcG9uZW50czoge1xuICAgIFZSYWRpb0dyb3VwLFxuICAgIFZSYWRpb1xuICB9XG59XG4iXX0=