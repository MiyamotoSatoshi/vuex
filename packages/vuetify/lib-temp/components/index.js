export * from './VApp';
export * from './VAlert';
export * from './VAutocomplete';
export * from './VAvatar';
export * from './VBadge';
export * from './VBottomNav';
export * from './VBottomSheet';
export * from './VBreadcrumbs';
export * from './VBtn';
export * from './VBtnToggle';
export * from './VCard';
export * from './VCarousel';
export * from './VCheckbox';
export * from './VChip';
export * from './VCombobox';
export * from './VCounter';
export * from './VDataIterator';
export * from './VDataTable';
export * from './VDatePicker';
export * from './VDialog';
export * from './VDivider';
export * from './VExpansionPanel';
export * from './VFooter';
export * from './VForm';
export * from './VGrid';
export * from './VHover';
export * from './VIcon';
export * from './VImg';
export * from './VInput';
export * from './VItemGroup';
export * from './VJumbotron';
export * from './VLabel';
export * from './VList';
export * from './VMenu';
export * from './VMessages';
export * from './VNavigationDrawer';
export * from './VOverflowBtn';
export * from './VPagination';
export * from './VPaper';
export * from './VParallax';
export * from './VPicker';
export * from './VProgressCircular';
export * from './VProgressLinear';
export * from './VRadioGroup';
export * from './VRangeSlider';
export * from './VRating';
export * from './VResponsive';
export * from './VSelect';
export * from './VSlider';
export * from './VSnackbar';
export * from './VSpeedDial';
export * from './VStepper';
export * from './VSubheader';
export * from './VSwitch';
export * from './VSystemBar';
export * from './VTabs';
export * from './VTextarea';
export * from './VTextField';
export * from './VTimeline';
export * from './VTimePicker';
export * from './VToolbar';
export * from './VTooltip';
export * from './VTreeview';
export * from './VWindow';
export * from './transitions';
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi9zcmMvY29tcG9uZW50cy9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxjQUFjLFFBQVEsQ0FBQTtBQUN0QixjQUFjLFVBQVUsQ0FBQTtBQUN4QixjQUFjLGlCQUFpQixDQUFBO0FBQy9CLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsVUFBVSxDQUFBO0FBQ3hCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsZ0JBQWdCLENBQUE7QUFDOUIsY0FBYyxnQkFBZ0IsQ0FBQTtBQUM5QixjQUFjLFFBQVEsQ0FBQTtBQUN0QixjQUFjLGNBQWMsQ0FBQTtBQUM1QixjQUFjLFNBQVMsQ0FBQTtBQUN2QixjQUFjLGFBQWEsQ0FBQTtBQUMzQixjQUFjLGFBQWEsQ0FBQTtBQUMzQixjQUFjLFNBQVMsQ0FBQTtBQUN2QixjQUFjLGFBQWEsQ0FBQTtBQUMzQixjQUFjLFlBQVksQ0FBQTtBQUMxQixjQUFjLGlCQUFpQixDQUFBO0FBQy9CLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsbUJBQW1CLENBQUE7QUFDakMsY0FBYyxXQUFXLENBQUE7QUFDekIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxVQUFVLENBQUE7QUFDeEIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxRQUFRLENBQUE7QUFDdEIsY0FBYyxVQUFVLENBQUE7QUFDeEIsY0FBYyxjQUFjLENBQUE7QUFDNUIsY0FBYyxjQUFjLENBQUE7QUFDNUIsY0FBYyxVQUFVLENBQUE7QUFDeEIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxhQUFhLENBQUE7QUFDM0IsY0FBYyxxQkFBcUIsQ0FBQTtBQUNuQyxjQUFjLGdCQUFnQixDQUFBO0FBQzlCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsVUFBVSxDQUFBO0FBQ3hCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMscUJBQXFCLENBQUE7QUFDbkMsY0FBYyxtQkFBbUIsQ0FBQTtBQUNqQyxjQUFjLGVBQWUsQ0FBQTtBQUM3QixjQUFjLGdCQUFnQixDQUFBO0FBQzlCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsU0FBUyxDQUFBO0FBQ3ZCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsZUFBZSxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSAnLi9WQXBwJ1xuZXhwb3J0ICogZnJvbSAnLi9WQWxlcnQnXG5leHBvcnQgKiBmcm9tICcuL1ZBdXRvY29tcGxldGUnXG5leHBvcnQgKiBmcm9tICcuL1ZBdmF0YXInXG5leHBvcnQgKiBmcm9tICcuL1ZCYWRnZSdcbmV4cG9ydCAqIGZyb20gJy4vVkJvdHRvbU5hdidcbmV4cG9ydCAqIGZyb20gJy4vVkJvdHRvbVNoZWV0J1xuZXhwb3J0ICogZnJvbSAnLi9WQnJlYWRjcnVtYnMnXG5leHBvcnQgKiBmcm9tICcuL1ZCdG4nXG5leHBvcnQgKiBmcm9tICcuL1ZCdG5Ub2dnbGUnXG5leHBvcnQgKiBmcm9tICcuL1ZDYXJkJ1xuZXhwb3J0ICogZnJvbSAnLi9WQ2Fyb3VzZWwnXG5leHBvcnQgKiBmcm9tICcuL1ZDaGVja2JveCdcbmV4cG9ydCAqIGZyb20gJy4vVkNoaXAnXG5leHBvcnQgKiBmcm9tICcuL1ZDb21ib2JveCdcbmV4cG9ydCAqIGZyb20gJy4vVkNvdW50ZXInXG5leHBvcnQgKiBmcm9tICcuL1ZEYXRhSXRlcmF0b3InXG5leHBvcnQgKiBmcm9tICcuL1ZEYXRhVGFibGUnXG5leHBvcnQgKiBmcm9tICcuL1ZEYXRlUGlja2VyJ1xuZXhwb3J0ICogZnJvbSAnLi9WRGlhbG9nJ1xuZXhwb3J0ICogZnJvbSAnLi9WRGl2aWRlcidcbmV4cG9ydCAqIGZyb20gJy4vVkV4cGFuc2lvblBhbmVsJ1xuZXhwb3J0ICogZnJvbSAnLi9WRm9vdGVyJ1xuZXhwb3J0ICogZnJvbSAnLi9WRm9ybSdcbmV4cG9ydCAqIGZyb20gJy4vVkdyaWQnXG5leHBvcnQgKiBmcm9tICcuL1ZIb3ZlcidcbmV4cG9ydCAqIGZyb20gJy4vVkljb24nXG5leHBvcnQgKiBmcm9tICcuL1ZJbWcnXG5leHBvcnQgKiBmcm9tICcuL1ZJbnB1dCdcbmV4cG9ydCAqIGZyb20gJy4vVkl0ZW1Hcm91cCdcbmV4cG9ydCAqIGZyb20gJy4vVkp1bWJvdHJvbidcbmV4cG9ydCAqIGZyb20gJy4vVkxhYmVsJ1xuZXhwb3J0ICogZnJvbSAnLi9WTGlzdCdcbmV4cG9ydCAqIGZyb20gJy4vVk1lbnUnXG5leHBvcnQgKiBmcm9tICcuL1ZNZXNzYWdlcydcbmV4cG9ydCAqIGZyb20gJy4vVk5hdmlnYXRpb25EcmF3ZXInXG5leHBvcnQgKiBmcm9tICcuL1ZPdmVyZmxvd0J0bidcbmV4cG9ydCAqIGZyb20gJy4vVlBhZ2luYXRpb24nXG5leHBvcnQgKiBmcm9tICcuL1ZQYXBlcidcbmV4cG9ydCAqIGZyb20gJy4vVlBhcmFsbGF4J1xuZXhwb3J0ICogZnJvbSAnLi9WUGlja2VyJ1xuZXhwb3J0ICogZnJvbSAnLi9WUHJvZ3Jlc3NDaXJjdWxhcidcbmV4cG9ydCAqIGZyb20gJy4vVlByb2dyZXNzTGluZWFyJ1xuZXhwb3J0ICogZnJvbSAnLi9WUmFkaW9Hcm91cCdcbmV4cG9ydCAqIGZyb20gJy4vVlJhbmdlU2xpZGVyJ1xuZXhwb3J0ICogZnJvbSAnLi9WUmF0aW5nJ1xuZXhwb3J0ICogZnJvbSAnLi9WUmVzcG9uc2l2ZSdcbmV4cG9ydCAqIGZyb20gJy4vVlNlbGVjdCdcbmV4cG9ydCAqIGZyb20gJy4vVlNsaWRlcidcbmV4cG9ydCAqIGZyb20gJy4vVlNuYWNrYmFyJ1xuZXhwb3J0ICogZnJvbSAnLi9WU3BlZWREaWFsJ1xuZXhwb3J0ICogZnJvbSAnLi9WU3RlcHBlcidcbmV4cG9ydCAqIGZyb20gJy4vVlN1YmhlYWRlcidcbmV4cG9ydCAqIGZyb20gJy4vVlN3aXRjaCdcbmV4cG9ydCAqIGZyb20gJy4vVlN5c3RlbUJhcidcbmV4cG9ydCAqIGZyb20gJy4vVlRhYnMnXG5leHBvcnQgKiBmcm9tICcuL1ZUZXh0YXJlYSdcbmV4cG9ydCAqIGZyb20gJy4vVlRleHRGaWVsZCdcbmV4cG9ydCAqIGZyb20gJy4vVlRpbWVsaW5lJ1xuZXhwb3J0ICogZnJvbSAnLi9WVGltZVBpY2tlcidcbmV4cG9ydCAqIGZyb20gJy4vVlRvb2xiYXInXG5leHBvcnQgKiBmcm9tICcuL1ZUb29sdGlwJ1xuZXhwb3J0ICogZnJvbSAnLi9WVHJlZXZpZXcnXG5leHBvcnQgKiBmcm9tICcuL1ZXaW5kb3cnXG5leHBvcnQgKiBmcm9tICcuL3RyYW5zaXRpb25zJ1xuIl19