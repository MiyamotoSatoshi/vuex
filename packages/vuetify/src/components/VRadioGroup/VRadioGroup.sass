@use './variables' as *

.v-radio-group
  grid-template-areas: "prepend label append" ". control ." "a messages b"

  > .v-label
    grid-area: $radio-group-label-grid-area
    margin-inline-start: $radio-group-label-margin-inline-start

    + .v-selection-control-group
      margin-top: $radio-group-label-selection-group-margin-top
