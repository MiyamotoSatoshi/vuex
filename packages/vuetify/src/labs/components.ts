export * from './VDataTable'
export * from './VVirtualScroll'
