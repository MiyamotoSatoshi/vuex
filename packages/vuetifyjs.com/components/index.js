import Page from './Page'
import PageHeading from './PageHeading'
import TextPage from './TextPage'

export default {
  Page,
  PageHeading,
  TextPage
}
