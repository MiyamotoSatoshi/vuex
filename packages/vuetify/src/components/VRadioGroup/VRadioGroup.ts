// Styles
import '../../styles/components/_selection-controls.sass'
import './VRadioGroup.sass'

// Extensions
import VInput from '../VInput'
import { BaseItemGroup } from '../VItemGroup/VItemGroup'

// Mixins
import Comparable from '../../mixins/comparable'

// Types
import mixins from '../../util/mixins'
import { PropValidator } from 'vue/types/options'

const baseMixins = mixins(
  Comparable,
  BaseItemGroup,
  VInput
)

/* @vue/component */
export default baseMixins.extend({
  name: 'v-radio-group',

  provide () {
    return {
      radioGroup: this
    }
  },

  props: {
    column: {
      type: Boolean,
      default: true
    },
    height: {
      type: [Number, String],
      default: 'auto'
    },
    name: String,
    row: Boolean,
    // If no value set on VRadio
    // will match valueComparator
    // force default to null
    value: {
      default: null
    } as PropValidator<any>
  },

  computed: {
    classes (): object {
      return {
        ...VInput.options.computed.classes.call(this),
        'v-input--selection-controls v-input--radio-group': true,
        'v-input--radio-group--column': this.column && !this.row,
        'v-input--radio-group--row': this.row
      }
    }
  },

  methods: {
    genDefaultSlot () {
      return this.$createElement('div', {
        staticClass: 'v-input--radio-group__input',
        attrs: {
          role: 'radiogroup'
        }
      }, VInput.options.methods.genDefaultSlot.call(this))
    },
    genInputSlot () {
      const render = VInput.options.methods.genInputSlot.call(this)

      delete render.data!.on!.click

      return render
    },
    onClick: BaseItemGroup.options.methods.onClick
  }
})
