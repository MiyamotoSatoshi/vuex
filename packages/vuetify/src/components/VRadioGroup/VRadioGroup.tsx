// Styles
import './VRadioGroup.sass'

// Components
import { filterControlProps } from '@/components/VSelectionControl/VSelectionControl'
import { filterInputProps, makeVInputProps, VInput } from '@/components/VInput/VInput'
import { makeSelectionControlGroupProps, VSelectionControlGroup } from '@/components/VSelectionControlGroup/VSelectionControlGroup'
import { VLabel } from '@/components/VLabel'

// Composables
import { IconValue } from '@/composables/icons'
import { useProxiedModel } from '@/composables/proxiedModel'

// Utilities
import { computed } from 'vue'
import { filterInputAttrs, genericComponent, getUid, omit, useRender } from '@/util'

// Types
import type { VInputSlots } from '@/components/VInput/VInput'
import type { VSelectionControlSlots } from '@/components/VSelectionControl/VSelectionControl'

export type VRadioGroupSlots = VInputSlots & VSelectionControlSlots

export const VRadioGroup = genericComponent<VRadioGroupSlots>()({
  name: 'VRadioGroup',

  inheritAttrs: false,

  props: {
    height: {
      type: [Number, String],
      default: 'auto',
    },

    ...makeVInputProps(),
    ...omit(makeSelectionControlGroupProps(), ['multiple']),

    trueIcon: {
      type: IconValue,
      default: '$radioOn',
    },
    falseIcon: {
      type: IconValue,
      default: '$radioOff',
    },
    type: {
      type: String,
      default: 'radio',
    },
  },

  emits: {
    'update:modelValue': (val: any) => true,
  },

  setup (props, { attrs, slots }) {
    const uid = getUid()
    const id = computed(() => props.id || `radio-group-${uid}`)
    const model = useProxiedModel(props, 'modelValue')

    useRender(() => {
      const [inputAttrs, controlAttrs] = filterInputAttrs(attrs)
      const [inputProps, _1] = filterInputProps(props)
      const [controlProps, _2] = filterControlProps({
        ...props,
        multiple: false as const,
      })
      const label = slots.label
        ? slots.label({
          label: props.label,
          props: { for: id.value },
        })
        : props.label

      return (
        <VInput
          class="v-radio-group"
          { ...inputAttrs }
          { ...inputProps }
          v-model={ model.value }
          id={ id.value }
        >
          {{
            ...slots,
            default: ({
              id,
              messagesId,
              isDisabled,
              isReadonly,
            }) => (
              <>
                { label && (
                  <VLabel id={ id.value }>
                    { label }
                  </VLabel>
                ) }

                <VSelectionControlGroup
                  { ...controlProps }
                  id={ id.value }
                  aria-describedby={ messagesId.value }
                  defaultsTarget="VRadio"
                  trueIcon={ props.trueIcon }
                  falseIcon={ props.falseIcon }
                  type={ props.type }
                  disabled={ isDisabled.value }
                  readonly={ isReadonly.value }
                  aria-labelledby={ label ? id.value : undefined }
                  { ...controlAttrs }
                  v-model={ model.value }
                  v-slots={ slots }
                />
              </>
            ),
          }}
        </VInput>
      )
    })

    return {}
  },
})

export type VRadioGroup = InstanceType<typeof VRadioGroup>
