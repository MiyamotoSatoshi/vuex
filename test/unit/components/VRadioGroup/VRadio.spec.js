import { test } from '@/test'
import { VRadioGroup, VRadio } from '@/components/VRadioGroup'

const warning = '[Vuetify] The v-radio component must be used inside a v-radio-group'
const error = 'TypeError: this.radio.register is not a function'

test('VRadio.vue', ({ mount }) => {
  it('should advise about v-radio-group being necessary', () => {
    mount(VRadio)

    expect(warning).toHaveBeenTipped()
  })

  // TODO: Enable test when there's a way to test $parent.$vnode.tag
  it('should not advise about v-radio-group being necessary in VRadioGroup', () => {
    const wrapper = mount(VRadioGroup, {
      slots: {
        default: [VRadio]
      }
    })
    // no expectation other than a lack of tip
  })

  it('should render role and aria-checked attributes on input group', () => {
    const wrapper = mount(VRadio, {
      data: {
        isActive: false
      },
      provide: {
        radio: {
          name: 'name',
          isMandatory: false
        }
      }
    })

    let inputGroup = wrapper.first('input')
    expect(inputGroup.getAttribute('role')).toBe('radio')
    expect(inputGroup.getAttribute('aria-checked')).toBe('false')

    wrapper.setData({ 'isActive': true })
    inputGroup = wrapper.first('input')
    expect(inputGroup.getAttribute('aria-checked')).toBe('true')
    expect(wrapper.html()).toMatchSnapshot()

    expect(error).toHaveBeenWarned()
  })

  it('should render aria-label attribute with label value on input group', () => {
    const wrapper = mount(VRadio, {
      propsData: {
        label: 'Test'
      },
      attrs: {},
      provide: {
        radio: {
          name: 'name',
          isMandatory: false
        }
      }
    })

    const inputGroup = wrapper.first('input')
    expect(inputGroup.getAttribute('aria-label')).toBe('Test')
    expect(wrapper.html()).toMatchSnapshot()

    expect(error).toHaveBeenWarned()
  })

  it('should not render aria-label attribute with no label value on input group', () => {
    const wrapper = mount(VRadio, {
      propsData: {
        label: null
      },
      provide: {
        radio: {
          name: 'name',
          isMandatory: false
        }
      }
    })

    const inputGroup = wrapper.first('input')
    expect(inputGroup.element.getAttribute('aria-label')).toBeFalsy()
    expect(wrapper.html()).toMatchSnapshot()

    expect(error).toHaveBeenWarned()
  })

  it('should render proper input name', () => {
    const wrapper = mount(VRadio, {
      provide: {
        radio: {
          name: 'name',
          isMandatory: false
        }
      }
    })

    const input = wrapper.find('input')[0]
    expect(input.getAttribute('name')).toBe('name')
    expect(wrapper.html()).toMatchSnapshot()

    expect(error).toHaveBeenWarned()
  })

  it('should register and unregister', () => {
    const register = jest.fn()
    const unregister = jest.fn()

    const wrapper = mount(VRadio, {
      attachToDocument: true,
      provide: {
        radio: {
          name: 'name',
          isMandatory: false,
          register: register,
          unregister: unregister
        }
      }
    })

    expect(register).toHaveBeenCalled()
    wrapper.destroy()
    expect(unregister).toHaveBeenCalled()
  })

  it('should not render ripple when ripple prop is false', () => {
    const wrapper = mount(VRadio, {
      propsData: {
        ripple: false
      },
      provide: {
        radio: {
          name: 'name',
          isMandatory: false
        }
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')

    expect(ripple).toHaveLength(0)

    expect(error).toHaveBeenWarned()
  })

  it('should render ripple when ripple prop is true', () => {
    const wrapper = mount(VRadio, {
      propsData: {
        ripple: true
      },
      provide: {
        radio: {
          name: 'name',
          isMandatory: false
        }
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    expect(ripple.element._ripple.enabled).toBe(true)
    expect(ripple.element._ripple.centered).toBe(true)

    expect(error).toHaveBeenWarned()
  })

  it('should toggle when space or enter is pressed', () => {
    const wrapper = mount(VRadio)

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    const input = wrapper.first('input')

    input.trigger('focus')
    input.trigger('keydown.enter')

    expect(change).toHaveBeenCalledTimes(1)

    input.trigger('keydown.space')
    expect(change).toHaveBeenCalledTimes(2)

    input.trigger('keydown.tab')
    expect(change).toHaveBeenCalledTimes(2)

    expect(wrapper.vm.isFocused).toBe(true)
    wrapper.vm.onBlur()
    expect(wrapper.vm.isFocused).toBe(false)

    expect(warning).toHaveBeenTipped()
  })

  it('should not generate own colors when parent is in error', async () => {
    const wrapper = mount(VRadio)

    expect(wrapper.vm.classes).toEqual({
      'theme--dark': false,
      'theme--light': true,
      'v-radio--is-disabled': false,
      'v-radio--is-focused': false
    })

    wrapper.setData({ isActive: true })

    await wrapper.vm.$nextTick()

    expect(wrapper.vm.classes).toEqual({
      'accent--text': true,
      'theme--dark': false,
      'theme--light': true,
      'v-radio--is-disabled': false,
      'v-radio--is-focused': false
    })

    expect(warning).toHaveBeenTipped()
  })

  it('should inject isMandatory', () => {
    const wrapper = mount(VRadio, {
      provide: {
        radio: {
          mandatory: true,
          register: jest.fn()
        }
      }
    })

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    expect(wrapper.vm.internalValue).toBe(undefined)

    wrapper.vm.isActive = true

    wrapper.vm.onChange()

    expect(change).not.toBeCalled()

    wrapper.vm.isActive = false

    wrapper.vm.onChange()

    expect(change).toBeCalled()
  })

  it('should use custom icons', () => {
    const wrapper = mount(VRadio, {
      propsData: {
        onIcon: 'foo',
        offIcon: 'bar'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setData({ isActive: true })

    expect(wrapper.html()).toMatchSnapshot()
    expect(warning).toHaveBeenTipped()
  })

  it('should check/uncheck the internal input', () => {
    const wrapper = mount(VRadio)

    expect(wrapper.vm.$refs.input.checked).toBe(false)

    wrapper.setData({ isActive: true })

    expect(wrapper.vm.$refs.input.checked).toBe(true)

    wrapper.setData({ isActive: false })

    expect(wrapper.vm.$refs.input.checked).toBe(false)
    expect(warning).toHaveBeenTipped()
  })

  it('should render themed component', () => {
    const wrapper = mount(VRadio, {
      propsData: {
        light: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(warning).toHaveBeenTipped()
  })
})
