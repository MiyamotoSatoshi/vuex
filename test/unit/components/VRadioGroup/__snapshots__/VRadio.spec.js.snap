// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VRadio.vue should not render aria-label attribute with no label value on input group 1`] = `

<div class="v-radio theme--light">
  <div class="v-input--selection-controls__input">
    <input aria-checked="false"
           role="radio"
           type="radio"
           name="name"
    >
    <div class="v-input--selection-controls__ripple">
    </div>
    <i aria-hidden="true"
       class="v-icon material-icons theme--light"
    >
      radio_button_unchecked
    </i>
  </div>
  <label aria-hidden="true"
         class="v-label theme--light"
         style="left: 0px; position: relative;"
  >
  </label>
</div>

`;

exports[`VRadio.vue should render aria-label attribute with label value on input group 1`] = `

<div class="v-radio theme--light">
  <div class="v-input--selection-controls__input">
    <input aria-label="Test"
           aria-checked="false"
           role="radio"
           type="radio"
           name="name"
    >
    <div class="v-input--selection-controls__ripple">
    </div>
    <i aria-hidden="true"
       class="v-icon material-icons theme--light"
    >
      radio_button_unchecked
    </i>
  </div>
  <label aria-hidden="true"
         class="v-label theme--light"
         style="left: 0px; position: relative;"
  >
    Test
  </label>
</div>

`;

exports[`VRadio.vue should render proper input name 1`] = `

<div class="v-radio theme--light">
  <div class="v-input--selection-controls__input">
    <input aria-checked="false"
           role="radio"
           type="radio"
           name="name"
    >
    <div class="v-input--selection-controls__ripple">
    </div>
    <i aria-hidden="true"
       class="v-icon material-icons theme--light"
    >
      radio_button_unchecked
    </i>
  </div>
  <label aria-hidden="true"
         class="v-label theme--light"
         style="left: 0px; position: relative;"
  >
  </label>
</div>

`;

exports[`VRadio.vue should render role and aria-checked attributes on input group 1`] = `

<div class="v-radio theme--light accent--text">
  <div class="v-input--selection-controls__input">
    <input aria-checked="true"
           role="radio"
           type="radio"
           name="name"
    >
    <div class="v-input--selection-controls__ripple accent--text">
    </div>
    <i aria-hidden="true"
       class="v-icon material-icons theme--light accent--text"
    >
      radio_button_unchecked
    </i>
  </div>
  <label aria-hidden="true"
         class="v-label theme--light"
         style="left: 0px; position: relative;"
  >
  </label>
</div>

`;

exports[`VRadio.vue should render themed component 1`] = `

<div class="v-radio theme--light">
  <div class="v-input--selection-controls__input">
    <input aria-checked="false"
           role="radio"
           type="radio"
    >
    <div class="v-input--selection-controls__ripple">
    </div>
    <i aria-hidden="true"
       class="v-icon material-icons theme--light"
    >
      radio_button_unchecked
    </i>
  </div>
  <label aria-hidden="true"
         class="v-label theme--light"
         style="left: 0px; position: relative;"
  >
  </label>
</div>

`;

exports[`VRadio.vue should use custom icons 1`] = `

<div class="v-radio theme--light">
  <div class="v-input--selection-controls__input">
    <input aria-checked="false"
           role="radio"
           type="radio"
    >
    <div class="v-input--selection-controls__ripple">
    </div>
    <i aria-hidden="true"
       class="v-icon material-icons theme--light"
    >
      bar
    </i>
  </div>
  <label aria-hidden="true"
         class="v-label theme--light"
         style="left: 0px; position: relative;"
  >
  </label>
</div>

`;

exports[`VRadio.vue should use custom icons 2`] = `

<div class="v-radio theme--light accent--text">
  <div class="v-input--selection-controls__input">
    <input aria-checked="true"
           role="radio"
           type="radio"
    >
    <div class="v-input--selection-controls__ripple accent--text">
    </div>
    <i aria-hidden="true"
       class="v-icon material-icons theme--light accent--text"
    >
      bar
    </i>
  </div>
  <label aria-hidden="true"
         class="v-label theme--light"
         style="left: 0px; position: relative;"
  >
  </label>
</div>

`;
